libs/cardano-ledger-core/src/Cardano/Ledger/Core.hs

```diff
@@ -463,6 +464,8 @@ class
 -- > type instance PreviousEra (AllegraEra c) = ShelleyEra c
 type family PreviousEra era = (r :: Type) | r -> era
 
+type family ByronTranslationContext era :: Type
```

Comment:
I see that you're instantiating this for eras other than `ShelleyEra`. That seems wrong to me. As far as I know, `ShelleyEra` is the only one that is designed to be capable of following `Byron`.

What do you think, @JaredCorduan ?

Reply:
Ah, I realize now that @bartfrenk copied these instances from my sketch in the Issue input-output-hk/ouroboros-network#3976. I suppose me disagreeing with my past self shows the sketch needs more thought!

In the Consensus layer, we're reusing one data type for all the Shelley-based eras. For the Shelley era itself, that data type should contain whatever it needs to translate the final Byron ledger state into the first Shelley ledger state. But we want any other era to have `()` for that field, since no other era needs to translate the final Byron ledger state.

Another way to think about it: the `TranslationContext` for `ShelleyEra` should include the necessary bits for the Byron ledger state translation, whereas all the era Shelley-based eras wouldn't need those bits. However, `TranslationContext` is only for translating from Shelley-based era to Shelley-based era, and thus there is no `TranslationContext` for `ShelleyEra` (since it's the first Shelley-based era).

So: `type instance ByronTranslationContext (BabbageEra _) = ()` doesn't mean "`BabbageEra` can translate from Byron without needing any addition data." It instead means "`BabbageEra` doesn't need any such context data since it never translates from Byron`".

Maybe a renaming would help clarify?

Alternatively, we could have this data type:

```
data ShelleyLedgerConfig era = ShelleyLedgerConfig {
      shelleyLedgerGlobals                     :: !SL.Globals
    , shelleyLedgerTranslationContext          :: !(Core.TranslationContext era)
    , shelleyLedgerTranslationContextFromByron :: !(Strict.Maybe (SL.ByronTranslationContext era))
    }

type instance ByronTranslationContext ShelleyEra = SL.ParedDownShelleyGenesis
type instance ByronTranslationContext AllegraEra = TypeError ...
type instance ByronTranslationContext AlonzoEra  = TypeError ...
type instance ByronTranslationContext BabbageEra = TypeError ...
type instance ByronTranslationContext ConwayEra  = TypeError ...
```

Thus the person constructing the `ShelleyLedgerConfig ` could `Just` only with the `ShelleyEra`, and must use `Nothing` with all the others, as intended. I _think_ this would also effectively prevent any of those non-`ShelleyEra` era's code from accessing this field: the type of the argument when checking a `Just x` pattern would raise a typeerror, wouldn't it?

Reply:
Actually, I think the following is the much simpler and ideal option. Just redefine the already existing `TranslationContext (ShelleyEra c)` type family instance from its current `()` right-hand side to the following:

```
-- | There is no 'TranslateEra (ShelleyEra c)' instance, since 'TranslateEra' is for translating from one Shelley-based era to the next. However, we do need extra data when translating from Byron to 'ShelleyEra' itself. Hence this instance.
type instance TranslationContext (ShelleyEra c) = TranslationContextByronToShelley c

data TranslationContextByronToShelley c = TranslationContextByronToShelley {
  byronToShelleyProtocolParams :: ShelleyPParams (ShelleyEra c)
, byronToShelleyGenDelegs :: GenDelegs c
, byronToShelleyMaxLovelaceSupply :: Word64
}```

cc @amesgen

Reply:
That indeed seems like the most minimal change that would do the job. :+1:

Obviously, this would special-case Byron as a predecessor of the first Shelley era, as we store the `TranslationContext` directly in the `ShelleyLedgerConfig` in Consensus. I am not sure how keen Ledger is on organizing the code in such a way that the Cardano chain (which includes Byron) is not given special treatment compared to a Shelley-only chain (where `TranslationContext (ShelleyEra c) ~ ()` would be natural) or even a chain where Shelley is preceded by a protocol other than Byron (on which `TranslationContext (ShelleyEra c)` might depend).

At least e.g. the docs of `ShelleyGenesis` are alluding to the possibility of a Shelley-only chain:
https://github.com/input-output-hk/cardano-ledger/blob/836452c1ff7b38b16ca5d2d3b0c58383ef59f512/eras/shelley/impl/src/Cardano/Ledger/Shelley/Genesis.hs#L121-L125

Reply:
@amesgen I believe that there are tests from the QA team, and benchmarks from the benchmarking, that make use of a Shelley-only chain. We should check with them.

> I am not sure how keen Ledger is on organizing the code in such a way that the Cardano chain (which includes Byron) is not given special treatment ...

I can't say that I understand the implications of this, sorry.